amundsen_application/models/user.py

```diff
@@ -1,120 +1,15 @@
-        self.last_name = last_name
-        self.manager_email = manager_email
-        self.manager_fullname = manager_fullname
-        self.profile_url = profile_url
-        self.role_name = role_name
-        self.slack_id = slack_id
-        self.team_name = team_name
-        self.user_id = user_id
-        # TODO: Add frequent_used, bookmarked, & owned resources
-
-
-class UserSchema(Schema):
-    display_name = fields.Str(allow_none=False)
-    email = fields.Str(required=True)
-    employee_type = fields.Str(allow_none=True)
-    first_name = fields.Str(allow_none=True)
-    full_name = fields.Str(allow_none=True)
-    github_username = fields.Str(allow_none=True)
-    is_active = fields.Bool(allow_none=True)
-    last_name = fields.Str(allow_none=True)
-    manager_email = fields.Str(allow_none=True)
-    manager_fullname = fields.Str(allow_none=True)
-    profile_url = fields.Str(allow_none=True)
-    role_name = fields.Str(allow_none=True)
-    slack_id = fields.Str(allow_none=True)
-    team_name = fields.Str(allow_none=True)
-    user_id = fields.Str(allow_none=False)
-
-    def _str_no_value(self, s: Optional[str]) -> bool:
-        # Returns True if the given string is None or empty
-        if not s:
-            return True
-        if len(s.strip()) == 0:
-            return True
-        return False
-
-    @pre_load
-    def preprocess_data(self, data: Dict) -> Dict:
-        if self._str_no_value(data.get('user_id')):
-            data['user_id'] = data.get('email')
-
-        if self._str_no_value(data.get('profile_url')):
-            data['profile_url'] = ''
-            if app.config['GET_PROFILE_URL']:
-                data['profile_url'] = app.config['GET_PROFILE_URL'](data['user_id'])
-
-        first_name = data.get('first_name')
-        last_name = data.get('last_name')
-
-        if self._str_no_value(data.get('full_name')) and first_name and last_name:
-            data['full_name'] = f"{first_name} {last_name}"
-
-        if self. _str_no_value(data.get('display_name')):
-            if self._str_no_value(data.get('full_name')):
-                data['display_name'] = data.get('email')
-            else:
-                data['display_name'] = data.get('full_name')
-
-        return data
-
-    @post_load
-    def make_user(self, data: Dict) -> User:
-        return User(**data)
-
-    @validates_schema
-    def validate_user(self, data: Dict) -> None:
-        if self._str_no_value(data.get('display_name')):
-            raise ValidationError('"display_name", "full_name", or "email" must be provided')
-
-        if self._str_no_value(data.get('user_id')):
-            raise ValidationError('"user_id" or "email" must be provided')
+from marshmallow import ValidationError
 
 
 def load_user(user_data: Dict) -> User:
     try:
         schema = UserSchema()
+        # To make sure we pass frontend configuration to amundsen common models
+        user_data['GET_PROFILE_URL'] = app.config['GET_PROFILE_URL']
```

Comment:
Just an FYI for this comment I left in amundsencommon, if we ever want top consider removing the need for this configuration -- which was initially another workaround. https://github.com/lyft/amundsencommon/pull/36/files#r425595217